src/components/WfsLayer.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react'
+import L, { GeoJSONOptions } from 'leaflet'
+import { FeatureCollection } from 'geojson'
+
+import { useMapInstance, GeoJSON } from '@datapunt/react-maps'
+import fetchWithAbort from '../utils/fetchWithAbort'
+import { ZoomLevel } from '../types'
+import { MIN_ZOOM_LEVEL, MAX_ZOOM_LEVEL } from '../constants'
+
+const NO_DATA: FeatureCollection = {
+  type: 'FeatureCollection',
+  features: [],
+}
+
+interface Props {
+  url: string
+  options?: GeoJSONOptions
+  zoomLevel: ZoomLevel
+}
+
+const getBBox = (mapInstance: any) => {
+  return `&${L.Util.getParamString({
+    bbox: mapInstance.getBounds().toBBoxString(),
+  }).substring(1)}`
+}
+
+const isVisible = (mapInstance: any, zoomLevel: ZoomLevel) => {
+  const { min, max } = zoomLevel
+  const zoom = mapInstance.getZoom()
+  return zoom <= (min || MIN_ZOOM_LEVEL) && zoom >= (max || MAX_ZOOM_LEVEL)
+}
+
+/**
+ * WfsLayer component should be only used when the api provides wfs capabilities for spatial queries
+ * The data is requested on each map action (pan, zoom)
+ *
+ */
+const WfsLayer: React.FC<Props> = ({ url, options, zoomLevel }) => {
+  const mapInstance = useMapInstance()
+  const [layerInstance, setLayerInstance] = useState()
+  const [bbox, setBbox] = useState('')
+  const [json, setJson] = useState<FeatureCollection>(NO_DATA)
+
+  if (mapInstance) {
+    mapInstance.on('moveend', () => {
+      const bb = getBBox(mapInstance)
+      setBbox(bb)
+    })
+  }
+
+  useEffect(() => {
+    if (!mapInstance) {
+      return () => {}
+    }
+
+    if (!isVisible(mapInstance, zoomLevel)) {
+      setJson(NO_DATA)
+      return
+    }
+
+    const extent = bbox || getBBox(mapInstance)
+    const [request, controller] = fetchWithAbort(`${url}${extent}`)
+
+    request
+      .then(res => res.json())
+      .then(res => setJson(res))
+      .catch(error => {
+        // Ignore abort errors since they are expected to happen.
+        if (error instanceof Error && error.name === 'AbortError') {
+          return
+        }
+
+        return Promise.reject(error)
+      })
+
+    return () => {
+      controller.abort()
+    }
+  }, [mapInstance, bbox])
+
+  useEffect(() => {
+    if (layerInstance) {
+      layerInstance.clearLayers()
+      layerInstance.addData(json)
+    }
+
+    return () => {
+      if (layerInstance) {
```

Comment:
Why do we need to do this? We didn't bind a click event..

Reply:
All the markers inside the layer get a click event for each `moveend`. This statement ensures that the click events are removed when the layer data is refreshed. I added a comment to explain.